src/internal.rs

```diff
@@ -373,19 +373,61 @@ pub type cdef_fn = Option<
 pub type cdef_dir_fn =
     Option<unsafe extern "C" fn(*const libc::c_void, ptrdiff_t, *mut libc::c_uint) -> libc::c_int>;
 
-pub type const_left_pixel_row = *const libc::c_void;
-pub type looprestorationfilter_fn = Option<
-    unsafe extern "C" fn(
-        *mut libc::c_void,
-        ptrdiff_t,
-        const_left_pixel_row,
-        *const libc::c_void,
-        libc::c_int,
-        libc::c_int,
-        *const LooprestorationParams,
-        LrEdgeFlags,
-    ) -> (),
->;
+pub type pixel_8bpc = uint8_t;
+pub type coef_8bpc = int16_t;
+
+pub type pixel_16bpc = uint16_t;
+pub type coef_16bpc = int32_t;
+
+pub type const_left_pixel_row<Pixel> = *const [Pixel; 4];
```

Comment:
I think we should avoid generics here until we do them more properly in #304.  The advantage of using it here is more minimal.

Reply:
Similarly to https://github.com/memorysafety/rav1d/pull/308/files#r1245938890, we can now move these definitions to inside of `trait BitDepth`.